docs/fundamentals/syslib-diagnostics/syslib1021.md

```diff
@@ -0,0 +1,15 @@
+---
+title: SYSLIB1021 error
+description: Learn about the diagnostic that generates compile-time error SYSLIB1021.
+ms.date: 05/07/2021
+---
+
+# SYSLIB1021: Can't have the same template placeholders with different casing
+
+A method annotated with the `LoggerMessageAttribute` attribute has multiple template placeholders that differ only by case.
```

Comment:
@maryamariyan I introduced the word "placeholder" here. Is that okay?

Reply:
how about "Can't have the same template argument with different casing"?

Reply:
where:
```
 "this is a message template with {param1} and {Param1}"
```

more on this: https://messagetemplates.org/

Reply:
So is it a "named hole" rather than a placeholder? I'm not sure I'd agree with argument, that would be a better term for the values passed into the second and third parameters here:

```
log("User {username} logged in from {ip_address}", username, ipAddress)
```

Would we call them parameters?

Reply:
"Named hole" isn't a term I've heard before, and sounds a little weird to me. But I'm willing to run with it if you think that's more appropriate than parameter or item name.